Adds Accordion component and patterns (#662)
microsoft/atlas-design: site/src/patterns/accordion.md

			</div>
		</div>
	</details>
	<details class="accordion accordion-xxl border-top">
		<summary class="padding-xs">
			<div class="accordion-header margin-left-xxs">
				<p class="font-weight-semibold">Accordion header 2 text</p>
				<p class="font-size-md color-text-subtle">Accordion subheader text</p>
			</div>
		</summary>
		<div class="accordion-content background-color-body-medium padding-xs">
			<article class="card card border-left-lg border-left-color-accent margin-bottom-xs">
				<div class="card-content">
					<p class="card-supertitle">Content Type</p>
					<a href="#" class="card-title">The headline of the content item</a>
					<p>Place card details items here.</p>
				</div>
			</article>
			<article class="card card border-left-lg border-left-color-accent">
				<div class="card-content">
					<p class="card-supertitle">Content Type</p>
					<a href="#" class="card-title">The headline of the content item</a>
					<p>Place card details items here.</p>
				</div>
			</article>
			<div class="text-align-center margin-top-xs">
				<a href="#" class="button button-sm button-primary">Button text</a>
			</div>
		</div>
	</details>
</div>
```

> ## Coda accordion

Author: ![image](https://github.com/user-attachments/assets/ef7d2014-d208-4492-b86e-39a897685036)

Was trying to think of a word signifying end, it's the pattern used for our FAQ page which is generally at the end of a page, the icon is aligned to the end of the component. Open to other names

> ## Coda accordion

Reviewer: I can't think of a word either. What do you think about calling it as it is? Accordion with right-aligned icon?

> ## Coda accordion

Reviewer 2: Gonna remove the patterns page for now. Designs have changed again so not a good time to settle on patterns.

> These accordions us standard `.accordion` subcomponents, border-atomics to create a . They use `.accordion-icon-end` to right align the icon. Background color atomics are used for the `.accordion-content` subcomponent. Due to the border and backgound color atomics used, a consistent spacing padding atomic is used to align the header and panel content. Note that is recommended right aligned icon accordions not be full-width elements at desktop sizes.

Reviewer: ```suggestion
These accordions use standard `.accordion` subcomponents and border-atomics to create an accordion . They use `.accordion-icon-end` to right align the icon. Background color atomics are used for the `.accordion-content` subcomponent. Due to the border and background color atomics used, a consistent spacing padding atomic is used to align the header and panel content. It is recommended that accordions with right aligned icons not be full-width elements at desktop sizes.
```

---
title: Accordion
description: Accordion patterns in the Atlas Design System
template: standard
# figmaEmbed: https://www.figma.com/design/a5gqz0qtUMAKF36lNS6b61/Adaptive-Learning---Phase-I-(2024)?node-id=2581-106784&t=lkaa1re80XFTF9O3-0
---

# Accordion patterns

> Because the `details` component has so many possible permutations, it can be helpful to provide examples of certain styles.

Reviewer 2: I wouldn't refer to details component but accordion instead. 

Because of the accordion component ....

Author: ✅Changed it to be more in line with the card component's reference to the card patterns page.

Reviewer 2: still see the `details` uses, please change to `accordion`

Author: This has been updated.

---
title: Accordion
description: Accordion patterns in the Atlas Design System
template: standard
---

# Accordion patterns

Because the `details` component has so many possible permutations, it can be helpful to provide examples of certain styles.

> ## Accordion

Reviewer 2: Can you provide a brief title and explanation for every pattern? Check the cards pattern page for the reference. Right now it's just a pile of code which kinda hard to go through and understand what are the scenarios

Author: Added titles for the accordion patterns

This accordion uses the `open` attribute to expand a designated panel on page load. Grouping `.accordion` components together using the `name` attribute, prevents more than 1 panel from being open at a time. Atomics classes are used to style the accordion `<details>` and `<summary>` elements.

```html
<details class="accordion background-color-body-accent border-radius margin-block-xxs">
	<summary class="font-weight-semibold padding-xs">
		<div class="accordion-header font-weight-semibold">Accordion headline 1 text</div>
	</summary>
	<div class="accordion-content padding-xs">
		<p class="margin-bottom-xxs">
			Viverra aliquet eget sit amet tellus cras. Nisl nisi scelerisque eu ultrices vitae.
			Condimentum id venenatis a condimentum. Ut eu sem integer vitae.
		</p>
		<h4 class="font-weight-semibold">Accordion content subheading</h4>
		<article class="card margin-top-xs">

> 			<div class="card-template">

Reviewer 2: same thing on all demo examples

				<p class="card-supertitle">Content Type</p>

> 				<a href="#" class="card-title margin-bottom-none">The headline of the content item</a>

Reviewer 2: same thing on all demo examples

	</div>
</details>
<details class="accordion background-color-body-accent border-radius margin-block-xxs">
	<summary class="padding-xs">
		<div class="accordion-header font-weight-semibold">Accordion headline 2 text</div>
	</summary>
	<div class="accordion-content padding-xs">
		<p class="margin-bottom-xxs">
			Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut
			labore et dolore magna aliqua. Ullamcorper velit sed ullamcorper morbi tincidunt ornare massa
			eget egestas.
		</p>
		<h4 class="font-weight-semibold">Accordion content subheading</h4>
		<article class="card margin-block-xs">
			<div class="card-template">
				<p class="card-supertitle">Content Type</p>
				<a href="#" class="card-title margin-bottom-none">The headline of the content item</a>
			</div>
		</article>
		<article class="card">
			<div class="card-template">
				<p class="card-supertitle">Content Type</p>
				<a href="#" class="card-title margin-bottom-none">The headline of the content item</a>
			</div>
		</article>
	</div>
</details>
```

```html
<details class="accordion accordion-lg border-top">
	<summary>
		<div class="accordion-header margin-left-xxs">

> 			<p class="font-weight-semibold font-size-h5 line-height-normal">Accordion header 1 text</p>

Reviewer 2: the sizing should be controlled by the accordion's sizing atomics. If there is a need of even a bigger size variation - it should be added to the component. Remove `font-size-h5 line-height-normal`

Reviewer 2: Revisit all examples

Author: Added xl and xxl large sizes and removed the font size atomics from the headers for this pattern

> 			<p class="font-size-md color-text-subtle">Accordion subheader text</p>

Reviewer 2: Adjust the font size atomic to match with `accordion-lg` from my previous comment. `font-size-sm`?

> 			<p class="font-size-md color-text-subtle">Accordion subheader text</p>

Reviewer 3: Two things about this nesting:

`<summary>`, having an implicit `button` role, will likely obliterate any semantics inside of it, meaning this `h3` will semantically not be treated as an `h3`. It might be worth just switching this to something more generic like a `<p>` and still applying the larger font size and whatnot - that way, we keep the presentation, but don't imply things like impacting heading-based navigation.

Secondly, every element in here gets collapsed into forming the `summary` button's accessible name - so in this case, the `summary` is named "Accordion header 1 text Accordion subheader text." Screenreaders will announce that in full, no differentiation between children, and in real world use cases, this could get quite verbose (especially when you throw in translation). Even worse, _some_ speech-dictation users would need to say this whole name in full to trigger the accordion: "Click Accordion header 1 text Accordion subheader text." This can also get really tedious in real world use cases.

I say this not to say that we _can't_ do this kind of subheading design, but rather, to suggest we minimize how often we do it, since it can introduce some heavy tediousness for some disabled users.

Author: Understood, this pattern was included because it's being used on the [Career Path](https://learn.microsoft.com/en-us/training/career-paths/ai-engineer), if we want to discourage it we don't need to document it in atlas, was just documenting the existing patterns we'll be replacing with this component to deprecate the _details.scss in docs-ui.

Want me to remove this pattern from atlas? Updated it to use `<p>` tags

Author: Added MDN link to this pattern about the button role's effects on semantics on L63
